src/main/java/org/openstreetmap/atlas/geography/boundary/CountryToShardListCache.java

```diff
@@ -0,0 +1,91 @@
+package org.openstreetmap.atlas.geography.boundary;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.geography.sharding.Sharding;
+import org.openstreetmap.atlas.geography.sharding.SlippyTile;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.streaming.writers.SafeBufferedWriter;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.maps.MultiMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class that stores country and intersecting shards in a {@link MultiMap}, and can be constructed
+ * from scratch or from file. Allows saving this mapping to file, which is useful when sharding is
+ * dense or borders are complex.
+ *
+ * @author james-gage
+ */
+public final class CountryToShardListCache
+{
+    private static final Logger logger = LoggerFactory.getLogger(CountryToShardListCache.class);
+    private static final String DELIMITER = "||";
+    private final MultiMap<String, SlippyTile> countryToShards = new MultiMap<>();
+
+    public CountryToShardListCache(final CountryBoundaryMap boundaries, final StringList countries,
```

Comment:
How about adding another constructor that uses all the countries in the boundarymap and dos not require a list of countries?